torch/csrc/jit/codegen/cuda/scheduler/registry.cpp

```diff
@@ -34,6 +34,35 @@ std::deque<std::deque<TensorView*>> tvChains(
   return tv_chains;
 }
 
+bool rejectScheduleForTorchGather(TorchGatherOp* torch_gather, 
+  ScheduleHeuristic schedule_stragety) {
+  if (!torch_gather->input(0)->isFusionInput()) {
+    scheduler_debug_utils::canScheduleRejectReason(
+      schedule_stragety,
+      "First input of TorchGatherOp must be fusion input.");
+      return true;
+  }
+  if (torch_gather->input(0)->uses().size() > 1) {
+    scheduler_debug_utils::canScheduleRejectReason(
+      schedule_stragety,
+      "First input of TorchGatherOp can only be used by TorchGatherOp");
+      return true;
+  }
+  if (!torch_gather->input(1)->isFusionInput()) {
+    scheduler_debug_utils::canScheduleRejectReason(
+      schedule_stragety,
+      "Index input of TorchGatherOp must be fusion input.");
```

Comment:
Nit. Error message is inconsistent. In the above it is mentioned `first input`, here we mention`Index input`.

Reply:
I will rename these.